core/primitives/piece/piece_data.hpp

```diff
@@ -40,4 +43,35 @@ namespace fc::primitives::piece {
     bool is_null_data_;
   };
 
+  class ReaderType {
+   public:
+    enum Type : uint64_t {
+      pushStreamReader = 0,
```

Comment:
[Enumerators should be named like constants](https://google.github.io/styleguide/cppguide.html#Enumerator_Names).
`kPushStreamReader`